### PR TITLE
fix(server): allow disabling content-type check

### PR DESCRIPTION
**Pull request by crenshaw-dev**
The current content type allowlist flag doesn't correctly handle the "none" (i.e. allow all) option. This PR adds support for that case.

I also added:
* warning log when checking is disabled
* argocd-cmd-params-cm support for the flag
* unit tests

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -993,6 +993,8 @@ func (a *ArgoCDServer) newHTTPServer(ctx context.Context, port int, grpcWebHandl
 	}
 	if len(a.ContentTypes) > 0 {
 		handler = enforceContentTypes(handler, a.ContentTypes)
+	} else {
```
Comment (todaywasawesome) [inline server/server.go:996]: @crenshaw-dev Is there a situation where someone could upgrade their manifests and the content types option is not set? I sort of think the if/else should be opposite handling, only when you expressly turn them off should we turn them off, other wise we default to them being on. You know what I mean?

**Review by todaywasawesome — CHANGES_REQUESTED** [review]

File: manifests/base/server/argocd-server-deployment.yaml [hunk 2]
```diff
@@ -251,6 +251,12 @@ spec:
                 name: argocd-cmd-params-cm
                 key: server.k8sclient.retry.base.backoff
                 optional: true
+        - name: ARGOCD_API_CONTENT_TYPES
+          valueFrom:
+            configMapKeyRef:
+              name: argocd-cmd-params-cm
+              key: server.api.content.types
+              optional: true
```
Comment (ishitasequeira) [inline manifests/base/server/argocd-server-deployment.yaml:259]: not: spacing seems to be off

**Review by ishitasequeira — CHANGES_REQUESTED** [review]

File: manifests/base/server/argocd-server-deployment.yaml (hunk 2 above)
Comment (crenshaw-dev) [reply manifests/base/server/argocd-server-deployment.yaml:259]: Fixed! The new block was okay but a bunch of others were indented by 4, which is weird.

File: server/server.go (hunk 1 above)
Comment (crenshaw-dev) [reply server/server.go:996]: Confirmed, if the user does not affirmatively set some value, the default is secure. To turn off validation, they must affirmatively set the allow list to an empty string.

**Review by todaywasawesome — APPROVED** [review]

**Review by ishitasequeira — APPROVED** [review]

**Comment by crenshaw-dev** [comment]
Manually opened cherry-pick PRs.

**Pull request merged**